type-definitions/combinator/errors.d.ts

```diff
@@ -0,0 +1,6 @@
+import { Stream } from '../types';
+
+export function recoverWith<A, B>(p: (a: B) => Stream<A>, s: Stream<A>): Stream<A>;
```

Comment:
Can TS constrain the bounds `B` to `Error`?  For example, [here's what recoverWith looks like in flow](https://github.com/mostjs/core/blob/master/src/index.js.flow#L242).  Also, might be a good visual cue to rename `B` -> `E` to indicate an error type.